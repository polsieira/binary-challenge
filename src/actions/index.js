export const checkIsLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
})

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
})

export const addExoplanets = (exoplanets) => ({
  type: 'ADD_EXOPLANETS',
  exoplanets
})

export const loginUser = (userInfo) => ({
  type: 'LOGIN_USER',
  userInfo
})

export const addFilters = (filters) => ({
  type: 'ADD_FILTERS',
  filters
})