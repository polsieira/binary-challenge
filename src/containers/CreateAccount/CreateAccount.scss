.CreateAccount {
  display: flex;
  flex-direction: column;
  height: 64vh;
  padding-top: 20px;
  text-align: center;
    
  .create-account-message {
    color: black;
    font-size: 18px;
    margin-top: 40px;;
  }

  .label {
    color: black;
  }

  .MuiFormControl-root {
    align-self: center;
    color: black;
    width: 50%;
  }

  .MuiButtonBase-root {
    align-self: center;
    background: #282727;
    color: white;
    height: 40px;
    margin-top: 20px;
    width: 50%;
  }
}