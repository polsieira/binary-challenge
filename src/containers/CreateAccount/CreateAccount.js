import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { hasErrored, checkIsLoading } from '../../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './CreateAccount.scss';
import PropTypes from 'prop-types';

export class SignUpForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      isLoggedIn: false
    };
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleClick = async () => {
    const { hasErrored } = this.props;
    try {
      //fetch user data
      //set user to store
      this.setState({ isLoggedIn: true });
      hasErrored('');
    } catch ({ message }) {
      hasErrored(message);
    }
    this.clearInputs();
  };

  clearInputs = () => {
    const stateKeys = Object.keys(this.state);
    stateKeys.forEach(stateKey =>
      this.setState({
        [stateKey]: ''
      })
    );
  };

  handleSubmit = event => {
    event.preventDefault();
    this.handleClick();
  };

  render() {
    if (this.state.isLoggedIn) {
      return <Redirect to='/' />;
    }
    const { name, email, password } = this.state;
    const { error } = this.props;
    return (
      <div className='Login'>
        <TextField
          type="name"
          hintText="Enter your name"
          floatingLabelText="Name"
          id='name'
          className={error ? 'form__input form__input--error' : 'form__input'}
          name='name'
          placeholder="John Smith"
          value={name}
          onChange={e => this.handleChange(e)}
        />
        <TextField
          id='email'
          type='text'
          placeholder='your@email.com'
          className={error ? 'form__input form__input--error' : 'form__input'}
          name='email'
          value={email}
          onChange={e => this.handleChange(e)}
          hintText="Enter your Username"
          floatingLabelText="Username"
        />
        <TextField
          type="password"
          hintText="Enter your Password"
          floatingLabelText="Password"
          id='password'
          className={error ? 'form__input form__input--error' : 'form__input'}
          name='password'
          placeholder='Must have at least 8 characters'
          value={password}
          onChange={e => this.handleChange(e)}
        />
        <RaisedButton label="Login" className='form__button' primary="true" onClick={(event) => this.handleClick(event)} />
      </div >
    );
  }
}

export const mapStateToProps = ({ error, isLoading }) => ({
  error,
  isLoading
});

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({ checkIsLoading, hasErrored }, dispatch);
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SignUpForm);

SignUpForm.propTypes = {
  error: PropTypes.string,
  hasErrored: PropTypes.func,
  isLoading: PropTypes.bool,
  checkIsLoading: PropTypes.func,
};