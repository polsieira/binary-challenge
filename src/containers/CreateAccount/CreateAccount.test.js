import React from 'react';
import { shallow } from 'enzyme';
import { CreateAccount, mapStateToProps, mapDispatchToProps } from './CreateAccount';
import { checkIsLoading, hasErrored, loginUser } from '../../actions';

jest.mock('../../apiCalls.js');

describe('CreateAccount', () => {
  describe('CreateAccount component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<CreateAccount error='' />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should match the snapshot when there is an error', () => {
      wrapper = shallow(<CreateAccount error='error' />);
      expect(wrapper).toMatchSnapshot();
    })

    it('should update the email and password in state when handleChange is called', () => {
      const mockEventName = {
        target: {
          name: 'name',
          value: 'Meep Meep'
        }
      };
      const mockEventEmail = {
        target: {
          name: 'email',
          value: 'email@email.com'
        }
      };
      const mockEventPassword = {
        target: {
          name: 'password',
          value: 'password'
        }
      };

      const expectedName = 'Meep Meep';
      const expectedEmail = 'email@email.com';
      const expectedPassword = 'password';

      wrapper.instance().handleChange(mockEventName);
      wrapper.instance().handleChange(mockEventEmail);
      wrapper.instance().handleChange(mockEventPassword);

      expect(wrapper.state('name')).toEqual(expectedName);
      expect(wrapper.state('email')).toEqual(expectedEmail);
      expect(wrapper.state('password')).toEqual(expectedPassword);
    });

    it('should clear all variables in state when clearInputs is invoked', () => {
      const expected = {
        name: '',
        email: '',
        password: '',
        isLoggedIn: ''
      };

      wrapper.instance().setState({
        name: 'Meep Meep',
        email: 'email@email.com',
        password: 'password'
      });

      wrapper.instance().clearInputs();

      expect(wrapper.state()).toEqual(expected);
    });

    it('should call handleClick on click', () => {
      wrapper.instance().handleClick = jest.fn();
      wrapper.instance().forceUpdate();

      wrapper.find('.form__button').simulate('click');

      expect(wrapper.instance().handleClick).toHaveBeenCalled();
    });

  });

  describe('mapStateToProps', () => {
    it('should return an object with a movies array, an error message and isLoading property', () => {
      const mockState = {
        error: '',
        isLoading: false
      };

      const expected = {
        error: '',
        isLoading: false
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {

    const mockDispatch = jest.fn();


  });

});