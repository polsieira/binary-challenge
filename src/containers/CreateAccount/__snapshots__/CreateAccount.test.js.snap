// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount CreateAccount component should match the snapshot 1`] = `
<div
  className="CreateAccount"
>
  <h2
    className="create-account-message"
  >
    Create account with name, email and password
  </h2>
  <WithStyles(ForwardRef(TextField))
    autoComplete="current-password"
    className="form__input"
    id="name"
    label="Name"
    margin="normal"
    name="name"
    onChange={[Function]}
    placeholder="John Smith"
    type="name"
    value=""
  />
  <WithStyles(ForwardRef(TextField))
    autoComplete="current-password"
    className="form__input"
    id="email"
    label="Username"
    margin="normal"
    name="email"
    onChange={[Function]}
    placeholder="your@email.com"
    type="text"
    value=""
  />
  <WithStyles(ForwardRef(TextField))
    autoComplete="current-password"
    className="form__input"
    id="password"
    label="Password"
    margin="normal"
    name="password"
    onChange={[Function]}
    placeholder="Must have at least 8 characters"
    type="password"
    value=""
  />
  <WithStyles(ForwardRef(Button))
    className="form__button"
    label="Login"
    onClick={[Function]}
    primary="true"
  >
    Create Account
  </WithStyles(ForwardRef(Button))>
</div>
`;

exports[`CreateAccount CreateAccount component should match the snapshot when there is an error 1`] = `
<div
  className="CreateAccount"
>
  <h2
    className="create-account-message"
  >
    <Component />
  </h2>
  <WithStyles(ForwardRef(TextField))
    autoComplete="current-password"
    className="form__input form__input--error"
    id="name"
    label="Name"
    margin="normal"
    name="name"
    onChange={[Function]}
    placeholder="John Smith"
    type="name"
    value=""
  />
  <WithStyles(ForwardRef(TextField))
    autoComplete="current-password"
    className="form__input form__input--error"
    id="email"
    label="Username"
    margin="normal"
    name="email"
    onChange={[Function]}
    placeholder="your@email.com"
    type="text"
    value=""
  />
  <WithStyles(ForwardRef(TextField))
    autoComplete="current-password"
    className="form__input form__input--error"
    id="password"
    label="Password"
    margin="normal"
    name="password"
    onChange={[Function]}
    placeholder="Must have at least 8 characters"
    type="password"
    value=""
  />
  <WithStyles(ForwardRef(Button))
    className="form__button"
    label="Login"
    onClick={[Function]}
    primary="true"
  >
    Create Account
  </WithStyles(ForwardRef(Button))>
</div>
`;
